docs/source/reference/comparison.rst

```diff
@@ -0,0 +1,178 @@
+.. _sky-compare:
+
+Comparing SkyPilot with other systems
+=====================================
+
+We are often asked "How does SkyPilot compare with XYZ?". Providing an unbiased and up-to date answer for such questions is not easy, especially when the differences may be qualitative.
```

Comment:
Kind of feel this para is a bit odd. Seems messages aren't lost if left out. Wdyt? Cc @Michaelvll for a look too.

Reply:
I added it because comparisons can be sensitive for the authors of other systems, especially if we do not present their strengths. I think this should stay in

Reply:
We can keep this, but maybe good to rephrase a bit of the sentence. Some reference for how other library starts the comparison:
* Terraform: https://developer.hashicorp.com/terraform/intro/vs
```
Because of this flexibility, Terraform can be used to solve many different problems. This means there are a number of existing tools that overlap with the capabilities of Terraform. We compare Terraform to a number of these tools, but it should be noted that Terraform is not mutually exclusive with other systems. It can be used to manage a single application, or the entire datacenter.
```
* Pulumi: https://www.pulumi.com/docs/concepts/vs/
```
Because of this broad array of supported scenarios, there are many tools that overlap with Pulumi’s capabilities. Many of these are complementary and can be used together, whereas some are “either or” decisions.
```
* Modin: https://modin.readthedocs.io/en/stable/getting_started/why_modin/modin_vs_dask_vs_koalas.html

Reply:
Reworded the paragraph. wdyt?